lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -967,12 +968,16 @@ def set_http_request(record, entry)
 
       latency = input.delete('latency')
       unless latency.nil?
-        # Parse latency to duration. If failed, skip setting latency.
-        duration_in_seconds = ChronicDuration.parse(latency)
-        if duration_in_seconds
+        # Parse latency. If no valid format is detected, skip setting latency.
+        # Format: space/tab (optional) + integer + point & decimal (optional)
+        #       + space/tab (optional) + "s" + space/tab (optional)
+        # e.g.: "1.42 s"
+        parsed_groups = latency.scan(LATENCY_REGEX)
+        if !parsed_groups.empty? && parsed_groups[0].length == 2
           # Split the integer and decimal parts in order to calculate seconds
           # and nanos.
-          (latency_seconds, decimal_part) = duration_in_seconds.divmod 1
+          latency_seconds = parsed_groups[0][0].to_i
```

Comment:
This might be a bit more readable if we change the regexp to `/^\s*(?<seconds>\d+)(?<decimal>\.\d+)?\s*s\s*$/`, and then use:
```
match = LATENCY_REGEX.match(latency)
if match
  latency_seconds = match['seconds'].to_i
  decimal_part = match['decimal'].to_f
  ...
end
```